### PR TITLE
ci: add macos to CI

### PR DESCRIPTION
**Pull request by addaleax**
##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

 ci

### CODE REVIEW

File: test/parallel/test-v8-serdes.js [hunk 1]
```diff
@@ -63,7 +65,7 @@ const deserializerTypeError =
 {
   const ser = new v8.DefaultSerializer();
   ser._writeHostObject = common.mustCall((object) => {
-    assert.strictEqual(object, process.stdin._handle);
+    assert.strictEqual(object, hostObject);
     const buf = Buffer.from('stdin');
```
Comment (hybrist) [inline test/parallel/test-v8-serdes.js]: [Curiosity / context question]: I assume that in the context of this test, `"stdin"` is meant to be just a random string constant. Would it be less confusing to change it to something like `"hostObjectTag"` or similar after this change? Or are you trying to keep the churn minimal?

Comment (addaleax) [reply test/parallel/test-v8-serdes.js]: No, I just did search/replace, that’s all. ;) Thanks for catching is, updated!

**Review by hybrist — APPROVED** [review]

File: test/parallel/test-v8-serdes.js [hunk 2]
```diff
@@ -20,6 +20,8 @@ const objects = [
   circular
 ];
 
+const hostObject = new (process.binding('js_stream').JSStream)();
```
Comment (Fishrock123) [inline test/parallel/test-v8-serdes.js:23]: 🤔 I thought this exploded last time I tried it.

**Review by Fishrock123 — APPROVED** [review]

File: test/parallel/test-v8-serdes.js (hunk 2 above)
Comment (addaleax) [reply test/parallel/test-v8-serdes.js:23]: You mean, constructing a `JSStream` like that? It seems to work fine to me

**Comment by addaleax** [comment]
Landed in 06a1e5412869f5d5bc289142e7fd896e3609e5ea, edc403ca54a685d818ae0fca31466c7439936d22

**Pull request closed without merging**